scripts/make-cluster-node-secret.sh

```diff
@@ -0,0 +1,226 @@
+while getopts ":n:c:fh" opt; do
+  case ${opt} in
+    h )
+      print_usage
+      exit 0
+      ;;
+    f )
+      FORCE_RECREATE="yes"
+      ;;
+    n )
+      NAMESPACE=${OPTARG}
+      ;;
+    c )
+      CONTAINER=${OPTARG}
+      ;;
+    : )
+      echo "Error: -${OPTARG} requires an argument." >&2
+      exit 1
+      ;;
+   \? )
+     echo "Invalid Option: -$OPTARG" 1>&2
+     echo "\n"
+     print_usage
+     exit 1
+     ;;
+  esac
+done
+shift $((OPTIND -1))
+CLUSTER=$1
+NODE_NAME=$2
+
+assert_cluster() {
+  local cluster=$1
+
+  printf "Checking cluster $cluster is configured correctly..." >&2
+  if [ ! -d "./clusters/$cluster/secrets" ]; then
+    echo -e "Cannot add secrets for cluster $cluster which does not exist" >&2
+    exit 1
+  fi
+  printf "OK\n" >&2
+}
+
+assert_not_node() {
+  local cluster=$1
+  local node_name=$2
+
+  printf "Checking node $node_name does not already exist in cluster $cluster..." >&2
+  if [ ! -z "$FORCE_RECREATE" ]; then
+    printf "SKIP\n" >&2
+  else
+    if [ -f "./clusters/$cluster/secrets/${node_name}_keys.yaml" ] ||
+       [ -f "./clusters/$cluster/secrets/${node_name}_keys.unc.yaml" ]; then
+      echo -e "Node $node_name already exists in cluster $cluster. Use -f to overrite anyway." >&2
+      exit 1
+    fi
+    printf "OK\n" >&2
+  fi
+}
+
+assert_namespace() {
+  local namespace=$1
+
+  printf "Checking that namespace $namespace is valid..." >&2
+  if [[ ! $namespace =~ ^[a-z0-9][-a-z0-9]{0,61}[a-z0-9]$ ]]; then
+    echo -e "$namespace is not a valid Kubernetes namespace" >&2
+    exit 1
+  fi
+  printf "OK'\n" >&2
+}
+
+assert_command() {
+  local command=$1
+
+  printf "Checking for presense of $command..." >&2
+  local path_to_executable=$(command -v ${command})
+
+  if [ -z "$path_to_executable" ] ; then
+    echo -e "Cannot find ${command} executable. Is it on your \$PATH?" >&2
+    exit 1
+  fi
+  printf "OK\n" >&2
+}
+
+pull_container() {
+  local container=$1
+
+  printf "Pulling container $container..." >&2
+  docker pull $container > /dev/null 2>&1
+  if [ $? -eq 0 ]; then
+    printf "OK\n" >&2
+  else
+    printf "FAIL\n" >&2
+    exit 1
+  fi
+}
+
+NODE_KEY=
+NODE_ID=
+generate_node_key() {
+  local container=$1
+
+  printf "Generating node-key..." >&2
+  if output=$(docker run --rm -t $container key generate-node-key 2>&1); then
+    printf "OK\n" >&2
+    output=(${output[@]})
+    NODE_ID=(${output[0]})
+    NODE_KEY=(${output[1]})
+  else
+    printf "FAIL\n" >&2
+    echo "$output" >&2
+    exit 1
+  fi
+}
+
+NODE_KEY=
+NODE_ID=
+generate_node_key() {
+  local container=$1
+  local output=
+
+  printf "Generating node-key..." >&2
+  if output=$(docker run --rm -t $container key generate-node-key 2>&1); then
+    printf "OK\n" >&2
+    output=(${output[@]})
+    NODE_ID=$(printf "${output[0]}" | tr -d '\r')
+    NODE_KEY=$(printf "${output[1]}" | tr -d '\r')
+  else
+    printf "FAIL\n" >&2
+    echo "$output" >&2
+    exit 1
+  fi
+}
```

Comment:
Duplicate function and variables?

Reply:
`generate_node_key` is defined twice I think?

Reply:
🤦‍♂️ , removed